tools/AutomationTestFramework/ManageScripts/ListTestFunctions.ps1

```diff
@@ -0,0 +1,74 @@
+﻿<#
+.SYNOPSIS
+Walks trought all test function headers and collects those 
+that has the 'AutomationTest' tag in the dot Description section
+.PARAMETER path
+Path to the directory where test sctipts are located
+.EXAMPLE
+ListTestFunctions "e:\git\azure-powershell\src\ResourceManager\Compute\Commands.Compute.Test\ScenarioTests"
+#>
+function ListTestFunctions(
+    [string] $path) {
+
+    $lines = [System.Collections.ArrayList]@()
+    
+    $currentFunctions = Get-ChildItem function:
+    $testFiles = Get-ChildItem -Path $path -Filter "*Tests.ps1" -ErrorAction Stop
+    $variableList = @()
+    
+    foreach ($testFile in $testFiles) {
+
+        # get function list from the script file
+        . "$path\$testFile"
+
+        $scriptFunctions = Get-ChildItem function: | Where-Object { $currentFunctions -notcontains $_ }
+        
+        $testFunctions = $scriptFunctions | Where-Object { 
+            ($_.Name.StartsWith("Test-", "CurrentCultureIgnoreCase")) 
+        } | Where-Object {
+            $desc = (Get-Help $_).Description
+            $desc -and ($desc[0].Text -contains 'AzureAutomationTest')
```

Comment:
Can we make the value we are searching for a parameter to this function, with defautl value 'AzureAutomationTest'?  This will enable us to add additional tags in the future.

Reply:
done.